shimmy/utils.py

```diff
@@ -0,0 +1,48 @@
+"""Utility functions for the compatibility wrappers."""
+from __future__ import annotations
+
+import copy
+from collections import OrderedDict
+from typing import Any
+
+import numpy as np
+from dm_env.specs import Array, BoundedArray, DiscreteArray
```

Comment:
We may want to put this under a file with a different name maybe, perhaps something like `utils/dm_env_utils.py`, under this convention, all the utils for the other wrappers will end up in utils and it'll be harder to segregate dependencies.